src/middlewares/cache.js

```diff
@@ -0,0 +1,38 @@
+const { createHash } = require('crypto')
+
+module.exports = (opts) => {
+  const defaultStorage = {}
+  const defaults = {
+    calculateCacheId: (event) => Promise.resolve(createHash('md5').update(JSON.stringify(event)).digest('hex')),
+    getValue: (key) => Promise.resolve(defaultStorage[key]),
+    setValue: (key, value) => {
+      defaultStorage[key] = value
+      return Promise.resolve()
+    }
+  }
+
+  const options = Object.assign({}, defaults, opts)
+  let currentCacheKey
+
+  return ({
+    before: (handler, next) => {
+      options.calculateCacheId(handler.event)
+        .then((cacheKey) => {
+          currentCacheKey = cacheKey
+          return options.getValue(cacheKey)
+        })
+        .then((cachedResponse) => {
+          if (cachedResponse) {
```

Comment:
can we change this with `if (cachedResponse === undefined) {`? . That way we gonna avoid false negatives when cache is returning `0`, `null` or empty string

Reply:
Good point :)